src/core/editform/qgsattributeeditorelement.cpp

```diff
@@ -22,14 +22,17 @@
 #include "qgsattributeeditorhtmlelement.h"
 #include "qgsattributeeditorqmlelement.h"
 #include "qgsattributeeditorrelation.h"
-
-
+#include "qgssymbollayerutils.h"
+#include "qgsfontutils.h"
 
 QDomElement QgsAttributeEditorElement::toDomElement( QDomDocument &doc ) const
 {
   QDomElement elem = doc.createElement( typeIdentifier() );
   elem.setAttribute( QStringLiteral( "name" ), mName );
   elem.setAttribute( QStringLiteral( "showLabel" ), mShowLabel );
+  elem.setAttribute( QStringLiteral( "labelColor" ), QgsSymbolLayerUtils::encodeColor( mLabelColor ) );
+  elem.appendChild( QgsFontUtils::toXmlElement( mLabelFont, doc, QStringLiteral( "labelFont" ) ) );
```

Comment:
There's a bit of a trap here -- see e.g. https://github.com/qgis/QGIS/pull/48322

If a user picks a font which doesn't have a native bold/italic style but then checks those options in the GUI, then the font will have bold/italic set via QFont::setBold/setItalic. This results in a "faux bold" / "faux italic" effect in Qt font rendering (which is fine here), BUT QgsFontUtils::toXmlElement ONLY stores the style name, which will be empty in these faux bold/italic cases. 

So the consequence is that the bold/italic setting will be lost in these cases. Also QgsFontUtils::toXmlElement doesn't store other font properties like underline or strikethrough, so those will also be lost here.

Reply:
What would you recommend? To store bold/italic etc. as separate properties?

Reply:
Yes, I think you'll need to store bold/italic/strike through/underline alongside the style name.